### PR TITLE
session-desktop-appimage: refactor - added desktop entry

### PR DESCRIPTION
**Pull request by Alexnortung**
###### Description of changes

This pull request adds an icon and desktop entry to `session-desktop-appimage`.

Related to #165808

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
Please do the same thing as ssb-patchwork
https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/ssb-patchwork/default.nix

File: pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix [hunk 1]
```diff
@@ -1,16 +1,42 @@
...
   };
 
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/share/applications/
+    cp -r bin $out/bin
+    cp ${icon} $out/share/applications/
+    cp ${./session-desktop.desktop} $out/share/applications/session-desktop.desktop
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix:36]: Please use makeDesktopItem.

File: pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix [hunk 2]
```diff
@@ -1,16 +1,42 @@
...
+  src = appimageTools.wrapType2 {
+    inherit version pname;
+    src = fetchurl {
+      url = "https://github.com/oxen-io/session-desktop/releases/download/v${version}/session-desktop-linux-x86_64-${version}.AppImage";
+      sha256 = "ca7754e59146633b71e66b02a90cff87e4f2574e57ff831ca4a5f983b7e2fbef";
+    };
   };
 
+  dontConfigure = true;
+  dontBuild = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix:30]: Does this break anything?

File: pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix [hunk 3]
```diff
@@ -1,16 +1,42 @@
...
-    url = "https://github.com/oxen-io/session-desktop/releases/download/v${version}/session-desktop-linux-x86_64-${version}.AppImage";
-    sha256 = "ca7754e59146633b71e66b02a90cff87e4f2574e57ff831ca4a5f983b7e2fbef";
+  pname = "session-desktop-appimage";
+  src = fetchFromGitHub {
+    owner = "oxen-io";
+    repo = "session-desktop";
+    rev = "v${version}";
+    sha256 = "mlTwlGDJHG5shyDvcCsdlj+RVIX/HXlN/aLowtHPHQw=";
+  };
+  icon = "${src}/images/session/session_icon.png";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix:17]:
```suggestion
```

File: pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix [hunk 4]
```diff
@@ -1,16 +1,42 @@
...
+    };
   };
 
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/share/applications/
+    cp -r bin $out/bin
+    cp ${icon} $out/share/applications/
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix:35]:
```suggestion
    cp ${src}/images/session/session_icon.png $out/share/applications/
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Alexnortung** [comment]
I changed it to use `appimageTools.extractType2` to get the image instead of fetching the entire repo from github.
Also using the `makeDesktopItem`function.

File: pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix [hunk 5]
```diff
@@ -1,15 +1,47 @@
...
+  };
+in
+stdenvNoCC.mkDerivation {
+  inherit version pname;
+  src = appimage;
+
+  installPhase = ''
+    mkdir -p $out/share/applications/
+    cp -r bin $out/bin
+    cp ${desktop-item}/share/applications/* $out/share/applications/
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix:43]: Please use copyDesktopItems.

**Pull request merged**